src/person.rb

```diff
@@ -1,21 +1,28 @@
 class Person
-  attr_reader :id
   attr_accessor :name, :age
+  attr_reader :id
+
+  def initialize(id, name = "Unknown", age = 0, parent_permission: true)
+    @id = id
 
-  def initialize(age:, name: 'unknown', parent_permission: true)
-    @id = Random.rand(1..10_000)
     @name = name
+
     @age = age
+
     @parent_permission = parent_permission
   end
 
   def can_use_services?
-    of_age? ? true : @parent_permission
+    if is_of_age? || @parent_permission
+      true
+    else
+      false
+    end
   end
```

Comment:
- [Optional] having `is_of_age? || @parent_permission` only should be enough here. as Ruby returns the last expression by default.

Reply:
Yes. Thans @TadesseDev I will look into it